headers/validation.h

```diff
@@ -280,15 +359,23 @@ struct BinomialDistributionGenerator {
 	long long n_;
 	double p_;
 
-	explicit BinomialDistributionGenerator(long long n, double p) : n_(n), p_(p) {}
+	explicit BinomialDistributionGenerator(long long n, double p) : n_(n), p_(p) {
+		assert(p_ >= 0);
+		assert(p_ <= 1);
+	}
 
 	// NOTE: Currently this retries instead of clamping to the interval.
 	T operator()(T low, T high, std::mt19937_64& rng) const {
-		T v;
-		do {
-			v = std::binomial_distribution<T>(n_, p_)(rng);
-		} while(v < low or v > high);
-		return v;
+		assert(low <= high);
+		// this will be slow for large n
```

Comment:
Hmm; that sounds like a regression. I don't think this is used much (at all) anyway, so it's fine. But maybe print a warning in the constructor when `n` is larger than 1000 or so